src/modules/esl-utils/misc/object.ts

```diff
@@ -33,13 +34,24 @@ export function defined<T>(...params: T[]) {
 
 /** Makes a flat copy without undefined keys */
 export function copyDefinedKeys<T>(obj?: T): Partial<T> {
+  return copy(obj || {}, (key, value) => value !== void 0);
+}
+
+/** Makes a plain copy of obj with properties satisfying the predicate
+ * If no predicate provided copies all own properties */
+export function copy<T>(obj: T, predicate: CopyPredicate  = () => true): Partial<T> {
```

Comment:
Super-cosmetic: but it's better to place generic implementation fist, and the consumers (omit, copyDefKeys) after